### DISCUSSION TITLE
Access to target branch on push

### DISCUSSION THREAD

**Discussion by solarmosaic-kflorence** [community/community#25813] [Actions]
<p>Is there any environment variable or context available that will identify the target branch for a push on a pull request? Eg. how would I set up a job that only runs for pushes on a PR with a specific target branch?</p>
<p>In the following example, the target branch would be “master”:</p>
<p><img src="https://user-images.githubusercontent.com/52473140/181093587-c96ff121-e165-488e-9ec2-20231e871ea9.png" alt="Screen Shot 2019-11-01 at 8.01.12 PM.png" data-base62-sha1="n013SFIUpAku0kk8LM7yskYY2BA" width="374" height="28"></p>

**Comment by solarmosaic-kflorence** [comment]
<p>By the way, it would also be useful to have access to diff (eg. a list of paths) generated for the action as described here: <a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#git-diff-comparisons" rel="nofollow noopener">https://help.github.com/en/github/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#git-diff-comparisons</a></p>

**Comment by solarmosaic-kflorence** [comment]
<p>Is what I want “base.ref” as shown here: <a href="https://developer.github.com/v3/activity/events/types/#pullrequestevent" rel="nofollow noopener">https://developer.github.com/v3/activity/events/types/#pullrequestevent</a> ?</p>
<p>If so, I will need access to this information in a push event.</p>

**Comment by ethomson** [comment, marked as answer]
<p>Base ref will give you the information about the base of a pull request - yes.  This information is provided to you in the <a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/contexts-and-expression-syntax-for-github-actions" rel="nofollow noopener">github context</a>.  So you can use this either as a variable in your workflow, or an environment variable in workflow executions.  For example:</p>
<pre><code>run: |
  echo "Base ref is $GITHUB_BASE_REF"
</code></pre>
<p>Or to use this as a test for a step:</p>
<pre><code>run: echo "Hello, world"
if: github.base_ref == 'refs/heads/master'
</code></pre>
<p>But if your goal is to create a <em>workflow</em> that only runs for pushes on a PR with a specific target branch then you can codify that in the trigger for the entire workflow:</p>
<pre><code>on:
  pull_request:
    branches:
    - master
</code></pre>

**Comment by solarmosaic-kflorence** [comment]
<p>My original intent was to be able to derive a diff list of files for changed on any PR which is set to merge into a specific branch. I want this to essentially be able to implement my own path filtering logic, similar to how you can restrict actions to specific repository paths, except that I want my action to apply to the entire repository but execute different commands based on the diff list. Your suggestion will allow me to do this, thank you (although it would be nice if Github also provided the list they diff’d against in the context).</p>
<p>One note: when using “pull_request” as the trigger instead of “push”, the SHA I get in ${{ github.sha }} does not seem to point to a valid commit. It seems to be referencing a commit on some internal pull request branch instead of the latest commit that triggered the pull request event (see: <a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/events-that-trigger-workflows#pull-request-event-pull_request" rel="nofollow noopener">https://help.github.com/en/github/automating-your-workflow-with-github-actions/events-that-trigger-workflows#pull-request-event-pull_request</a>) – that doesn’t seem very useful, how can I reference the commit that caused the trigger when using this event type?</p>

**Comment by solarmosaic-kflorence** [comment]
<p>Ah, the following works for me:</p>
<pre><code>${{ github.event.pull_request.head.sha }}
</code></pre>
<p>A bit confusing but it’s in there. Thanks for the tips!</p>

**Comment by mishakorablin** [comment]
<p>Came across the same issue. Wouldn’t using <code>pull_request</code> event only trigger during specific actions (<a href="https://developer.github.com/v3/activity/events/types/#pullrequestevent" rel="nofollow noopener">https://developer.github.com/v3/activity/events/types/#pullrequestevent</a>)? So if I open PR and then push more commits into that branch it doesn’t seem that the event will fire for those pushes.</p>

**Comment by solarmosaic-kflorence** [comment]
<p><a class="mention" href="/mkorablin">@mkorablin</a> you can specify which PR events you care about: <a href="https://help.github.com/en/github/automating-your-workflow-with-github-actions/events-that-trigger-workflows#pull-request-event-pull_request" rel="nofollow noopener">https://help.github.com/en/github/automating-your-workflow-with-github-actions/events-that-trigger-workflows#pull-request-event-pull_request</a></p>
<p>By default, if you don’t specify any types, it will use  <strong>opened</strong> ,  <strong>synchronize</strong> , or  <strong>reopened</strong>. It’s not very clear from the documentation, but  <strong>synchronize</strong>  gets triggered when updates are pushed to the PR (eg: a new commit is pushed).</p>
